src/lib/menu/menu.md

```diff
@@ -93,6 +93,46 @@ that should trigger the sub-menu:
 
 <!-- example(nested-menu) -->
 
+### Lazy rendering
+By default, the menu content will be initialized even when the panel is closed. To defer
+initialization until the menu is open, the content can be provided as an `ng-template`
+with the `matMenuContent` attribute:
+
+```html
+<mat-menu #appMenu="matMenu">
```

Comment:
What do you think of avoiding the extra `matMenuContent` by letting people add `matMenu` to an `ng-template` element? Something like
```html
<ng-template matMenu>
  ...
</ng-template>
```
The selector for the menu would change to `mat-menu, ng-template[matMenu]`. The menu would then optionally inject `TemplateRef` and go through the extra steps if one is present.

Reply:
That's what I was going for initially, but Angular throws an error when you have a component on an `ng-template`. The alternative would be to use a directive, but we can't do that either, because we need to wrap the menu content in a div to add the proper styling and for the animation to work.